src/ui/UserInterface.java

```diff
@@ -545,10 +546,177 @@ public void getMemberInfo() {
 		}
 	}
 
+	/**
+	 * Takes from the user a member ID, a starting date, an ending date
+	 * and finds all transactions with given member ID between given dates.
+	 */
 	public void printTransactions() {
+		boolean sentinel = false;
+		String userInput = "";
+		String memberId = "";
+		Result member = null;
+		Calendar startingDate = Calendar.getInstance();
+		Calendar endingDate = Calendar.getInstance();
+		SimpleDateFormat dateFormat = new SimpleDateFormat("MM/dd/yyyy");
+		
+		//Loop the opportunity to input member ID until valid member ID is found or the user exits.
+		do {
+			System.out.print("Input member ID: ");
```

Comment:
I think we need to explicitly say that the user can enter "END" with each printout from the system.

Also I think there is a get string method already built in? Just to account for the user giving invalid inputs

Reply:
There is, but Eclipse complained that it "wasn't sure" what I was calling .equals() on was a String or not. The toString() can be removed, but I always assume an IDE has a better handle on code safety than myself.

Reply:
Oh no, I mean a get string method made by pavil in the UserInterface. Search for getString in the ui

Reply:
Oh yeah didn't see that. I'll change how I handle input.